Ginger/Ginger/Repository/RepositoryItemFactory.cs

```diff
@@ -744,8 +744,16 @@ public void DownloadSolution(string SolutionFolder)
             {
                 sol.ExistInLocaly = true;
             }
-            else if (WorkSpace.Instance.UserProfile.SourceControlType == SourceControlBase.eSourceControlType.GIT && Directory.Exists(PathHelper.GetLongPath(sol.LocalFolder + @"\.git")))
+            else if (WorkSpace.Instance.UserProfile.SourceControlType == SourceControlBase.eSourceControlType.GIT)
             {
+                if (!sol.LocalFolder.EndsWith(".git"))
+                {
```

Comment:
Directory.Exists() lines are not part of the IF so i'm not getting how it is checking if git folder is exist or not and setting sol.ExistInLocaly flag accordinally